lib/std/crypto/25519/ed25519.zig

```diff
@@ -51,7 +58,7 @@ pub const Ed25519 = struct {
             var h = Sha512.init(.{});
             h.update(&ss);
             h.final(&az);
-            const p = try Curve.basePoint.clampedMul(az[0..32].*);
+            const p = Curve.basePoint.clampedMul(az[0..32].*) catch return error.IdentityElement;
```

Comment:
Why did you change this?

Reply:
The multiplication doesn't assume a trusted base, and can return `WeakPublicKey`.

But a new key pair is being created here. Since we are multiplying the base point, the result is always in the main group; we are never going to compute a weak key.

`IdentityElement` was already the only possible error that could be returned.

The `catch` was just added to match the function error set. There's no behavior change here.

Reply:
To not mess up return traces, shouldn't we rethrow the original error?

Reply:
> To not mess up return traces, shouldn't we rethrow the original error?

How would you do that?

`catch |e| return @errSetCast(IdentityElementError, e);` works, but I don't feel comfortable using `@errSetCast` in case `clampedMul()` later returns something that's not in the `IdentityElementError` set.

That being said, the error means that the argument to this function was zero. Traces of internal functions are not super useful.